### PR TITLE
Integrate Sygma bridge into thala&rhala

### PR DESCRIPTION
**Pull request by tolak**
## Description

This PR contains stuff of integrating [Sygma](https://github.com/sygmaprotocol/sygma-substrate-pallets) bridge into SubBridge. The way we integrate Sygma pallets into our runtime is we introduced a dedicated wrapper pallet that implements `BridgeTransact` trait by calling `SygmaBridge.deposit` extrinsic under the hood, which means we don't change any single line code of Sygma pallet.

For now, Sygma bridge is only enabled on Thala runtime, after Sygma relayer confirms they can execute proposals on our blockchain, we will enable it on Rhala.

## New interfaces

- AssetRegisty.enable_sygmabridge
- AssetRegistry.disable_sygmabridge

## Sygma path spec

When using SubBridge to transfer assets to EVM chain from our chain, `dest` should be:

```
{"parents": 0, "interior": {"X3": [{"GeneralKey": "sygma"}, {"GeneralIndex": sygma_dest_domain_id} {"GeneralKey": {...account_address_on_ethereum}}]}}
```

When using SubBridge(khala) to transfer assets to EVM chain from other parachain, `dest` should be:

```
{"parents": 1, "interior": {"X4": [{"Parachain": 2004}, {"GeneralKey": "sygma"}, {"GeneralIndex": sygma_dest_domain_id} {"GeneralKey": {...account_address_on_ethereum}}]}}
```

### CODE REVIEW

**Comment by tolak** [comment]
The integration test error `Call enum size should be smaller than 1024 bytes` thrown by the substrate will be fixed after [this PR](https://github.com/sygmaprotocol/sygma-substrate-pallets/pull/67) is merged in Sygma pallets repo

**Review by h4x3rotab — APPROVED** [review]
lgtm

File: pallets/subbridge/src/sygma_wrapper.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
+pub use self::pallet::*;
+
+#[allow(unused_variables)]
+#[allow(clippy::large_enum_variant)]
+#[frame_support::pallet]
+pub mod pallet {
+	use crate::traits::*;
+	use assets_registry::SYGMA_PATH_KEY;
+	use frame_support::{dispatch::RawOrigin, pallet_prelude::*, transactional};
+	use funty::Fundamental;
```
[the remark this answers is not in the document]
Comment (tolak) [reply pallets/subbridge/src/sygma_wrapper.rs:10]: It being used when we convert u128 domain id to u8 with as-casted

File: pallets/assets-registry/src/lib.rs [hunk 2]
```diff
@@ -820,17 +1091,74 @@ pub mod pallet {
...
+
+	// Return (asset_id, asset decimal) pair from registried assets
+	impl<T: Config> Get<Vec<(XcmAssetId, u8)>> for Pallet<T> {
+		fn get() -> Vec<(XcmAssetId, u8)> {
+			let mut decimals: Vec<(XcmAssetId, u8)> =
+				vec![(Concrete(T::NativeAssetLocation::get()).into(), 12)];
+
+			// Lookup RegistryInfoByIds find all assets' decimal
+			let _: Vec<()> = RegistryInfoByIds::<T>::iter()
+				.map(|(_, info)| {
```
[the remark this answers is not in the document]
Comment (tolak) [reply pallets/assets-registry/src/lib.rs]: Done

File: runtime/khala/src/lib.rs [hunk 3]
```diff
@@ -1941,6 +1948,13 @@ impl_runtime_apis! {
         }
     }
 
+    impl sygma_runtime_api::SygmaBridgeApi<Block> for Runtime {
+		fn is_proposal_executed(nonce: DepositNonce, domain_id: DomainID) -> bool {
+            // TODO: enable Sygma
```
Comment (freddyli7) [inline runtime/khala/src/lib.rs:1953]: @tolak just out of curiosity, here for phala and khala, you didn't impl the api call body, but in thala runtime you impl it, do you plan to add this when later you guys going to launch on Kusama and Polkadot? or there is some other reason?

Comment (tolak) [reply runtime/khala/src/lib.rs:1953]: @freddyli7 Yes, will implement it when we enable Sygma on Kusama and Polkadot parachain

**Pull request merged**